Document k6 cloud run --local-execution (#1732)
grafana/k6-docs: docs/sources/next/results-output/real-time/cloud.md

test but you'll of course only have access to as much data as was streamed
before the network issue.

> {{< admonition type="caution" >}}
> 
> To make it possible for us to offer you a seamless user experience, the `k6 cloud run --local-execution script.js`
> command will also upload the test archive k6 produces from your test script and resources to our cloud service.
> 
> This behavior can be deactivated by adding the `--no-archive-upload` option to your
> command: `k6 cloud run --local-execution --no-archive-upload script.js`.
> 
> {{< /admonition >}}
> 
> {{< admonition type="note" >}}
> 
> {{< /admonition >}}

Reviewer: ```suggestion
{{< admonition type="caution" >}}

The `k6 cloud run --local-execution script.js` command uploads the test archive k6 produces from your test script and resources to Grafana Cloud.

This behavior can be deactivated by adding the `--no-archive-upload` option to your
command: `k6 cloud run --local-execution --no-archive-upload script.js`.

{{< /admonition >}}
```

Is there any reason why a user might want to disable this or keep it enabled? I think we should include it here as well if we want to use the `caution` admonition type. Otherwise, we can just switch it to `note`.

Author: A user might want to use the `--no-archive-upload` option primarily for privacy reasons. A user might also use the `k6 cloud run—-local-execution` mode specifically to keep all their context, assets, and resources local while still benefiting from the Grafana Cloud k6 experience.

It's activated by default because uploading the archive to the cloud, like with the `cloud run` command by default, gives the Grafana Cloud k6 much more context about the execution and leads to a better UX now and in the future.

I'll commit your suggestion as-is; cheers for that, and I'll keep the "caution" state as I think users who care about that do **really** care about it, and we should ensure they don't miss that bit.

Reviewer: That makes sense, thanks for the explanation! 🙇
